netlink-packet-wireguard/src/nlas/allowedip.rs

```diff
@@ -0,0 +1,80 @@
+use crate::{constants::*, raw::*};
+use anyhow::Context;
+use byteorder::{ByteOrder, NativeEndian};
+use libc::{in6_addr, in_addr};
+use netlink_packet_utils::{
+    nla::{Nla, NlaBuffer},
+    parsers::*,
+    traits::*,
+    DecodeError,
+};
+use std::{
+    mem::{size_of, size_of_val},
+    net::IpAddr,
+};
+
+#[derive(Clone, Debug, PartialEq, Eq)]
+pub enum WgAllowedIpAttrs {
+    Unspec(Vec<u8>),
+    Family(u16),
+    IpAddr(IpAddr),
```

Comment:
It seems that the netlink packet contains a socket address (with the port, adress family, and a couple additional fields for IPv6). So is there a reason for not making this a `SocketAddr` instead? (I've changed it to `SocketAddr` in my PR but maybe that was wrong).

Reply:
This is the Wireguard's allowed IPs. Each entry represents a CIDR IP network that is allowed to pass through the Wireguard interface. 
It is not related to the netlink socket address.

Reply:
Ah ok. Sorry I messed this up in https://github.com/Leo1003/netlink/pull/1. I'll try to update my PR tonight.

Reply:
Done